### DISCUSSION TITLE
let copilot decides which files needed to put in context.

### DISCUSSION THREAD

**Discussion by hamidrezaabroshan** [community/community#129149] [Copilot Conversations]
### Select Topic Area

Question

### Body

I have been using ChatGPT to perform repetitive coding tasks since the GPT-3.5 beta launch. Constantly copying and pasting pieces of code was a frustrating experience for me.
I purchased GitHub Copilot and have been using it for two months. However, I have not had a good experience in terms of context awareness. I still prefer to copy and paste code into GPT and then copy the result back into my code. Even trying to mention different files with `#file:filename` has been difficult and annoying. The `@workspace` command did not solve my problem either, and I feel something is missing here.
I had an idea that by allowing users to ask questions directly to Copilot and sharing the current method or method definition, GitHub Copilot should automatically determine which parts of the file or the entire file are needed. An LLM should decide which other files are necessary, and language tools like Java tools in VSCode should locate these files and put them in context. Users should be able to guide the LLM with commands such as "grab ProductDto and all its subclasses." or "write a method like payOnce."
We can grab needed info by using function calling of llm, searching feature of IDE,   a local small language model or using ide's language tools and parsers to give correct context to the llm.
Additionally, the LLM can determine which irrelevant parts of the discussion should be removed or archived to maximize context length.
Furthermore, if all the classes and methods could be shown in a tree view, users would be able to see the selected classes and methods and change them easily. This feature would significantly improve the user experience by providing a clear and organized overview of the relevant code.
Lastly, Copilot should share the folder structure or provide a checkbox to allow users to choose whether to share the folder structure or not. This option would give users more control over what information is shared and help in maintaining context.

**Comment by Akash1134** [comment]
We appreciate your feedback and your valuable input. We want to be transparent about our current priorities. You can gain insights into the projects we’re currently focusing on for the upcoming year by checking our [public roadmap](https://github.com/orgs/github/projects/4247/views/1). While smaller fixes may not always be explicitly listed, they often get addressed when related work is in progress.

Regarding your specific request, it’s important to note that it’s not actively being worked on at the moment. However, this doesn’t mean we won’t consider it in the future; it’s simply not on our development roadmap right now.

We’d like to emphasize that as a valued GitHub user, your feedback in this Discussion space, along with appropriate tags and examples, plays a crucial role in shaping the development of GitHub. While we can’t guarantee immediate action on every piece of feedback, please rest assured that we are attentive to your suggestions and continuously strive to enhance the GitHub experience.
While we do review all incoming feedback unfortunately due to the volume of feedback, we may not be able to respond to each one individually at this time. However, we strongly encourage you to continue providing your valuable input as it greatly helps us prioritize and improve GitHub.

Your contributions are instrumental in making GitHub better for all users. Thank you for taking the time to share your thoughts with us, even if the particular request is not currently in active development. Your input is truly valuable and contributes to the continuous improvement of GitHub.
